File: Semana_2/Playwright/TypeScript/npm/curso/e2e/pages/separated/menu.actions.ts
import { Page } from '@playwright/test';
import { MenuLocators } from './menu.locators';
import { capture } from '../evidence';

export class MenuActions {
  private readonly page: Page;
  private readonly locators: MenuLocators;

  constructor(page: Page) {
    this.page = page;
    this.locators = new MenuLocators(page);
  }

  async openNewAccount(): Promise<void> {
    await this.locators.openNewAccountLink().click();
    await capture(this.page, 'parabank-separated-open-new-account');
  }
}

File: Semana_2/Selenium/Java/maven/curso/e2e/src/test/java/com/curso/selenium/pages/separated/MenuActions.java
package com.curso.selenium.pages.separated;

import com.curso.selenium.evidence.EvidenceSupport;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class MenuActions {
    private final WebDriver driver;
    private final WebDriverWait wait;

    public MenuActions(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    public void openNewAccount() {
        wait.until(ExpectedConditions.elementToBeClickable(MenuLocators.OPEN_NEW_ACCOUNT_LINK)).click();
        EvidenceSupport.capture(driver, "parabank-separated-open-new-account");
    }
}
